substrate/frame/support/src/traits/misc.rs

```diff
@@ -36,6 +36,20 @@ pub const DEFENSIVE_OP_PUBLIC_ERROR: &str = "a defensive failure has been trigge
 #[doc(hidden)]
 pub const DEFENSIVE_OP_INTERNAL_ERROR: &str = "Defensive failure has been triggered!";
 
+/// Trait to get the number of variants in any enum.
+///
+/// NOTE: can be removed once <https://doc.rust-lang.org/std/mem/fn.variant_count.html> is stable.
+pub trait VariantCount {
+	/// Get the number of variants.
+	fn variant_count() -> u32;
```

Comment:
can this be `const`? the nightly function is

Reply:
Yep changed to const now.